src/netlink_nlctrl.c

```diff
@@ -0,0 +1,259 @@
+/*
+ * Copyright (c) 2017-2024 The strace developers.
+ * All rights reserved.
+ *
+ * SPDX-License-Identifier: LGPL-2.1-or-later
+ */
+
+#include "defs.h"
+#include "netlink_generic.h"
+#include "nlattr.h"
+#include <linux/cgroupstats.h>
+#include <linux/devlink.h>
+#include <linux/ethtool_netlink.h>
+#include <linux/ioam6_genl.h>
+#include <linux/mptcp_pm.h>
+#include <linux/netdev.h>
+#include <linux/nl80211.h>
+#include <linux/seg6_genl.h>
+#include <linux/tcp_metrics.h>
+#include <linux/thermal.h>
+
+#include "xlat/genl_ctrl_cmd.h"
+#include "xlat/genl_ctrl_attr.h"
+#include "xlat/genl_ctrl_attr_op.h"
+#include "xlat/genl_ctrl_attr_op_flags.h"
+#include "xlat/genl_ctrl_attr_mcast_grp.h"
+#include "xlat/genl_ctrl_attr_policy.h"
+#include "xlat/genl_devlink_cmd.h"
+#include "xlat/genl_ethtool_msg_send.h"
+#include "xlat/genl_ethtool_msg_recv.h"
+#include "xlat/genl_ioam6_cmd.h"
+#include "xlat/genl_mptcp_pm_cmd.h"
+#include "xlat/genl_netdev_cmd.h"
+#include "xlat/genl_nl80211_cmd.h"
+#include "xlat/genl_seg6_cmd.h"
+#include "xlat/genl_taskstats_cmd.h"
+#include "xlat/genl_tcp_metrics_cmd.h"
+#include "xlat/genl_thermal_cmd.h"
+#include "xlat/nl_attr_type.h"
+#include "xlat/nl_policy_type_attr.h"
+
+#define ARG_PAIR(arg) { arg, arg }
+static const struct {
+	const char *family;
+	const struct xlat *cmd[2];
+} family_names[] = {
+	{ "nlctrl", ARG_PAIR(genl_ctrl_cmd) },
```

Comment:
We can unify `decoders` in lookup_genl_decoder() in src/netlink_generic.c and `family_names`.

Reply:
Here is my version: https://github.com/masatake/strace/blob/b82b7a73e8ebb2f16c0ae42424616e75ef5b3495/src/netlink_generic.h#L15 .

If the direction is acceptable, I will make a new pull request after you merge this pull request.

Reply:
The only complication I see is testing because strace will be able to resolve identifiers only for those genl families that are registered in the kernel.  How do we test decoding of other genl families?